File: Harish Kumar/Dynamic Programming & Recursion/Count Number of Teams/Count Number of Teams.java
class Solution {
    public int numTeams(int[] rating) {
        
        int total = 0;
        
        for(int i = 0; i < rating.length;i++)
        {
            int leftGreater = 0, leftSmaller = 0;
            int rightGreater = 0, rightSmaller = 0;
            
            for(int j = 0; j < i; j++)
            {
                if(rating[j] < rating[i])
                {
                    leftSmaller++;
                }
                else
                {
                    leftGreater++;
                }
            }
            
            for(int j = i+1; j < rating.length;j++)
            {
                if(rating[j] > rating[i])
                {
                    rightGreater++;
                }
                else
                {
                    rightSmaller++;
                }
            }
            
            total += leftSmaller*rightGreater + leftGreater*rightSmaller;
        }
        return total;

    }
}

File: Shreya/Recursion and DP/Count Number of Teams.cpp
class Solution {
public:
    int numTeams(vector<int>& rating) {
        
        int ans = 0;
        for (int i = 1; i < rating.size()-1; i++) {
            int leftLarger = 0, leftSmaller = 0, rightLarger = 0, rightSmaller = 0;
            for (int j = 0; j < i; j++)
                if (rating[j] < rating[i])
                    leftSmaller++;
                else if (rating[j] > rating[i])
                    leftLarger++;
            for (int j = i + 1; j < rating.size(); j++)
                if (rating[j] < rating[i])
                    rightSmaller++;
                else if (rating[j] > rating[i])
                    rightLarger++;
            ans += leftSmaller * rightLarger + leftLarger * rightSmaller;
        }
        return ans;
    }
};